From 393b3170f45588c3ddcfd86e0abb3b22ba085ffc Mon Sep 17 00:00:00 2001
Message-Id: <393b3170f45588c3ddcfd86e0abb3b22ba085ffc.1369221838.git.minovotn@redhat.com>
In-Reply-To: <44d9306ede26ea2881e6c1f104b901be647022a8.1369221838.git.minovotn@redhat.com>
References: <44d9306ede26ea2881e6c1f104b901be647022a8.1369221838.git.minovotn@redhat.com>
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 14 May 2013 11:11:10 +0200
Subject: [PATCH 3/8] qdev: New qdev_prop_set_string()

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1368529872-8973-2-git-send-email-kraxel@redhat.com>
Patchwork-id: 51350
O-Subject: [RHEL-6.5 qemu-kvm PATCH 1/3] qdev: New qdev_prop_set_string()
Bugzilla: 947416
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit cc98467327e13adca8f65b5a841c08930ee68220)

Conflicts:

	hw/qdev-properties.c
---
 hw/qdev-properties.c |    5 +++++
 hw/qdev.h            |    1 +
 2 files changed, 6 insertions(+)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/qdev-properties.c | 5 +++++
 hw/qdev.h            | 1 +
 2 files changed, 6 insertions(+)

diff --git a/hw/qdev-properties.c b/hw/qdev-properties.c
index d47b213..3c47a98 100644
--- a/hw/qdev-properties.c
+++ b/hw/qdev-properties.c
@@ -720,6 +720,11 @@ void qdev_prop_set_uint64(DeviceState *dev, const char *name, uint64_t value)
     qdev_prop_set(dev, name, &value, PROP_TYPE_UINT64);
 }
 
+void qdev_prop_set_string(DeviceState *dev, const char *name, char *value)
+{
+    qdev_prop_set(dev, name, &value, PROP_TYPE_STRING);
+}
+
 int qdev_prop_set_drive(DeviceState *dev, const char *name, BlockDriverState *value)
 {
     int res;
diff --git a/hw/qdev.h b/hw/qdev.h
index 43899df..9574da1 100644
--- a/hw/qdev.h
+++ b/hw/qdev.h
@@ -299,6 +299,7 @@ void qdev_prop_set_uint16(DeviceState *dev, const char *name, uint16_t value);
 void qdev_prop_set_uint32(DeviceState *dev, const char *name, uint32_t value);
 void qdev_prop_set_int32(DeviceState *dev, const char *name, int32_t value);
 void qdev_prop_set_uint64(DeviceState *dev, const char *name, uint64_t value);
+void qdev_prop_set_string(DeviceState *dev, const char *name, char *value);
 void qdev_prop_set_chr(DeviceState *dev, const char *name, CharDriverState *value);
 void qdev_prop_set_netdev(DeviceState *dev, const char *name, VLANClientState *value);
 void qdev_prop_set_vlan(DeviceState *dev, const char *name, VLANState *value);
-- 
1.7.11.7

